gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/ReplicationDataRetentionCategory.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.replication;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.typesafe.config.Config;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+
+/**
+ * Specify the replication data type: sync, finite_snapshot,  finite_daily_partition or finite_hourly_partition
+ * <ul>
+ *  <li>sync: all files in source need to be copied to target
+ *  <li>finite_daily_partition : only limited daily partitions in source need to be copied to target
+ *  <li>finite_daily_partition : only limited hourly partitions in source need to be copied to target
+ *  <li>finite_snapshot: only limited snapshot directories in source need to be copied to target
+ * </ul>
+ * @author mitu
+ *
+ */
+@AllArgsConstructor
+public class ReplicationDataRetentionCategory {
+  enum Type{
```

Comment:
Why do we need the Type enum instead of implementations of the same interface? Can't this functionality be part of the source dataset implementation?